Gateway/Request/RefundDataBuilder.php

```diff
@@ -168,7 +168,7 @@ public function build(array $buildSubject): array
             );
 
             if ($this->adyenHelper->isPaymentMethodOpenInvoiceMethod($brandCode)) {
-                $openInvoiceFieldsCreditMemo = $this->openInvoiceHelper->getOpenInvoiceData($creditMemo->getOrder());
+                $openInvoiceFieldsCreditMemo = $this->openInvoiceHelper->getOpenInvoiceDataFromPayment($payment);
```

Comment:
What would you think about using `getCreditmemosCollection()` for refunds? That allows us to get correct items. Last refunded items might be different than last invoiced items.

Reply:
I've inject the credit memo invoice, to be use instead of the last invoice

```
$openInvoiceFieldsCreditMemo = $this->openInvoiceHelper->getOpenInvoiceDataFromPayment($payment, $creditMemo->getInvoice());
```